test/core/pipeline/features/conftest.py

```diff
@@ -1,11 +1,11 @@
-from typing import Tuple, List, Dict, Any
+from typing import Tuple, List, Dict, Any, Set
 
 from pytest_bdd import when, then, parsers
 
 from haystack import Pipeline
 
 
-PipelineData = Tuple[Pipeline, List[Dict[str, Any]], List[Dict[str, Any]], List[List[str]]]
+PipelineData = Tuple[Pipeline, List[Tuple[Dict[str, Any], Set[str]]], List[Dict[str, Any]], List[List[str]]]
 PipelineResult = Tuple[List[Dict[str, Any]], List[Dict[str, Any]], List[List[str]], List[List[str]]]
```

Comment:
This is very unwieldy - We should switch to dataclasses if we have more then two elements in an unnamed tuple.

Reply:
This will fixed a later PR. 👍